src/vs/editor/contrib/hover/browser/resizableContentWidget.ts

```diff
@@ -0,0 +1,122 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ResizableHTMLElement } from 'vs/base/browser/ui/resizable/resizable';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { ContentWidgetPositionPreference, ICodeEditor, IContentWidget, IContentWidgetPosition } from 'vs/editor/browser/editorBrowser';
+import { EditorOption } from 'vs/editor/common/config/editorOptions';
+import { clamp } from 'vs/base/common/numbers';
+import { IPosition, Position } from 'vs/editor/common/core/position';
+import * as dom from 'vs/base/browser/dom';
+
+const HEADER_HEIGHT = 30;
+const MIN_HEIGHT = 24;
+
+export abstract class ResizableContentWidget extends Disposable implements IContentWidget {
+
+	readonly allowEditorOverflow: boolean = true;
+	readonly suppressMouseDown: boolean = false;
+
+	protected readonly _resizableNode = this._register(new ResizableHTMLElement());
+	protected _contentPosition: IContentWidgetPosition | null = null;
+	protected _position: Position | undefined;
+
+	private _isResizing: boolean = false;
+
+	constructor(
+		protected readonly _editor: ICodeEditor,
+		_initialSize: dom.IDimension = new dom.Dimension(10, 10)
+	) {
+		super();
+		this._resizableNode.domNode.style.position = 'absolute';
+		this._resizableNode.minSize = new dom.Dimension(10, 10);
+		this._resizableNode.enableSashes(true, true, true, true);
+		this._resizableNode.layout(_initialSize.height, _initialSize.width);
+		this._register(this._resizableNode.onDidResize(e => {
+			this._resize(new dom.Dimension(e.dimension.width, e.dimension.height));
+			if (e.done) {
+				this._isResizing = false;
+			}
+		}));
+		this._register(this._resizableNode.onDidWillResize(() => {
+			this._isResizing = true;
+		}));
+	}
+
+	get isResizing() {
+		return this._isResizing;
+	}
+
+	abstract getId(): string;
+
+	getDomNode(): HTMLElement {
+		return this._resizableNode.domNode;
+	}
+
+	getPosition(): IContentWidgetPosition | null {
+		return this._contentPosition;
+	}
+
+	get position(): Position | undefined {
+		return this._position;
+	}
+
+	set position(position: Position | undefined) {
+		this._position = position;
+	}
```

Comment:
suggestion move this up into the ContentWidget because it seems specific to the content hover and not to resizable hovers in general